lib/ansible/cli/doc.py

```diff
@@ -443,6 +441,13 @@ def post_process_args(self, options):
 
         display.verbosity = options.verbosity
 
+        if options.json_format and options.show_snippet:
+            # --snippet and --json are incompatible
+            raise AnsibleOptionsError("Conflicting options used, only one of --snippet or --json can be used at the same time.")
+
+        if options.entry_point is not None and options.type not in ('role',):
+            raise AnsibleOptionsError("Option --entry-point can only be used with --type=role.")
```

Comment:
the mutually exclusive group should already do this

Reply:
There isn't a type-specific mutually exclusive group, and --json is not part of the mutually exclusive group either. Fixing these TODO's is part of the goal of https://github.com/ansible/ansible/pull/77464 (a draft, and in the process of being rewritten).

Reply:
this line confused me:
` exclusive = self.parser.add_mutually_exclusive_group()`

but now i remember the conversation around that PR

Reply:
@Akasurde the issue with this is the same as https://github.com/ansible/ansible/pull/77035#discussion_r835464915. So for example, --entry-point should show up as an option for -t role rather than as a common option for all types, and should be an error during argument parsing (rather than after).